### PR TITLE
change: [M3-6522] - Select components stories cleanup

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
Relatively small tech debt PR that does not affect the client facing code.

## Changes  🔄
- Remove `.mdx` story in favor of a `.tsx` one
- Mark 'Select/EnhancedSelect' as deprecated to avoid new usage
- Add new story for `<RegionSelect />`
- Group all "select" stories in storybook

## Preview 📷

Screenshot 2023-11-14 at 15 39 11

### Prerequisites
- Pull code locally and run `yarn storybook`

### Verification steps 
- Navigate to http://localhost:6006/?path=/docs/components-selects-enhanced-select--documentation and confirm changes & behavior of RegionSelect and EnhancedSelect stories

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/EnhancedSelect/Select.tsx [hunk 1]
```diff
@@ -67,22 +67,54 @@ export interface BaseSelectProps<
...
+   */
   hideLabel?: boolean;
+  /**
+   * If true, the label will be displayed inline.
+   * @default false
+   */
   inline?: boolean;
+  /**
+   * To specify a custom input ID
+   * @default false
```
Comment (bnussman-akamai) [inline packages/manager/src/components/EnhancedSelect/Select.tsx:111]: Optional:

Not sure how to notate this, but inputId defaults to `convertToKebabCase(label)` unless it is explicitly defined.

File: packages/manager/src/components/EnhancedSelect/Select.tsx [hunk 2]
```diff
@@ -67,22 +67,54 @@ export interface BaseSelectProps<
...
+   */
   inputId?: any;
```
Comment (bnussman-akamai) [inline packages/manager/src/components/EnhancedSelect/Select.tsx:113]: Optional: replace `any` with `string`. (This causes one other type error that would have to be addressed)

```suggestion
  inputId?: string;
```

File: packages/manager/src/components/EnhancedSelect/Select.tsx [hunk 3]
```diff
@@ -67,22 +67,54 @@ export interface BaseSelectProps<
   Clearable extends boolean = false
 > extends Omit<SelectProps<I, IsMulti>, 'onChange'>,
     CreatableSelectProps<I, IsMulti> {
+  /**
+   * Optional additional class names to apply to the root element.
+   */
   classes?: any;
```
Comment (bnussman-akamai) [inline packages/manager/src/components/EnhancedSelect/Select.tsx:73]: I don't see `classes` being used anywhere within this component. Maybe we can remove it?

**Review by bnussman-akamai — APPROVED** [review]
Nice organization! 🧹

 All of my feedback can be optional because it's beyond the scope of this ticket

File: packages/manager/src/components/EnhancedSelect/EnhancedSelect.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+const meta: Meta<BaseSelectProps<Item, boolean, boolean>> = {
+  args: {
+    isClearable: true,
+    isMulti: false,
+    label: 'Fruit',
+    options: fruits,
+  },
+  component: Select,
+  title: 'Components/Selects/Enhanced Select',
```
Comment (coliu-akamai) [inline packages/manager/src/components/EnhancedSelect/EnhancedSelect.stories.tsx:90]: optional nitpick, but maybe we could put something like `(deprecated)` in the title as well? that way anyone looking at the selects would know this is deprecated from the start:
image


```suggestion
  title: 'Components/Selects/Enhanced Select (deprecated)',
```

**Review by coliu-akamai — APPROVED** [review]
Awesome, thanks Alban! 
✅ confirmed select stories + behavior

File: packages/manager/src/components/EnhancedSelect/Select.tsx (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/src/components/EnhancedSelect/Select.tsx]: good call - i added it as such, which is ok i think

File: packages/manager/src/components/EnhancedSelect/Select.tsx (hunk 2 above)
Comment (abailly-akamai) [reply packages/manager/src/components/EnhancedSelect/Select.tsx]: cool, typed this better

File: packages/manager/src/components/EnhancedSelect/Select.tsx (hunk 3 above)
Comment (abailly-akamai) [reply packages/manager/src/components/EnhancedSelect/Select.tsx:73]: removed!

File: packages/manager/src/components/EnhancedSelect/EnhancedSelect.stories.tsx (hunk 4 above)
Comment (abailly-akamai) [reply packages/manager/src/components/EnhancedSelect/EnhancedSelect.stories.tsx]: Good idea! done

**Pull request merged**